// @smt-ui/component/src/voice-recognizer/src/const.js
/**
 * @file 常量
 * @author v_zhaoxinggang
 */

// 提示文案
export const TIPS = {
    DEFAULT_TEXT: '正在聆听中…',
    RETRY: '没听清，请重新尝试',
    TRY_AGIN: '出错了，请重新尝试',
    TOO_SHORT: '说话时间太短',
    TOO_LONG: '说话时间太长',
    FAIL: '识别失败',
    ABNORMAL_MIKE: '麦克风异常'
};

export const BTN_TEXT = {
    START: '点击说话',
    END: '说完了'
};

export const JUSTIFY_CONTENT = {
    MORE: 'flex-end',
    NORMAL: 'top'
};

export const PERM_TIPS = {
    TITLE: '开启麦克风权限',
    CONTENT: '1.点击小程序右上角更多，在"设置-权限管理"中开启麦克风权限\n2.在手机设置页中允许百度APP使用麦克风权限',
    BTN_TEXT: '我知道了'
};

// 动画图片
export const RIPPLE_GIF = [
    'https://b.bdstatic.com/searchbox/icms/searchbox/img/ripple-init.gif',
    'https://b.bdstatic.com/searchbox/icms/searchbox/img/ripple-voice.gif',
    'https://b.bdstatic.com/searchbox/icms/searchbox/img/ripple-err.gif'
];

// 错误码提示文案映射
export const ERR_MSG_MAP = {
    2001: TIPS.ABNORMAL_MIKE,
    2002: TIPS.RETRY,
    2003: TIPS.TOO_SHORT,
    1001: TIPS.TRY_AGIN,
    2009: TIPS.TRY_AGIN,
    3001: TIPS.TRY_AGIN,
    3002: TIPS.TRY_AGIN,
    9000: TIPS.TRY_AGIN,
    4002: TIPS.TOO_LONG,
    4001: TIPS.FAIL,
    4003: TIPS.FAIL,
    4004: TIPS.FAIL,
    2004: '',
    2005: ''
};
